webpack/test-utils/react-testing-lib-wrapper.js

```diff
@@ -0,0 +1,39 @@
+// For react-testing-library helpers, overrides, and utilities
+// See https://testing-library.com/docs/react-testing-library/setup for more info
+import React from 'react';
+import thunk from 'redux-thunk';
+import Immutable from 'seamless-immutable';
+import { render } from '@testing-library/react';
+import { createStore, applyMiddleware, combineReducers } from 'redux';
+import { Provider } from 'react-redux';
+import { Router } from 'react-router-dom';
+import { createMemoryHistory } from 'history';
+
+// Renders testable component with redux and react-router according to our usage
+function katelloRender(
+  component,
+  {
+    namespace, // redux namespace
+    reducer,
+    initialState = {},
+    route = '/',
+    history = createMemoryHistory({ initialEntries: [route] }),
+  } = {},
+) {
+  // Adding the reducer in the expected namespaced format
+  const combinedReducers = combineReducers({ katello: combineReducers({ [namespace]: reducer }) });
+  // Namespacing the initial state as well
+  const initialKatelloState = Immutable({ katello: { [namespace]: initialState } });
+  const store = createStore(combinedReducers, initialKatelloState, applyMiddleware(thunk));
```

Comment:
The middleware applied here should be the same as in real Foreman, not just thunk

Reply:
I'm trying to use the middleware from `foremanReact` for example:
```
import * as foremanMiddleware from 'foremanReact/redux/middlewares'
```
But it looks like `foremanReact` isn't available in a testing environment, we [mock it in other places](https://github.com/Katello/katello/blob/6a8d63b972dbbadfe1974690273d33ebc9e35a18/webpack/components/Content/Details/__tests__/ContentDetailInfo.test.js#L7)

Any thoughts on how to proceed?

Reply:
Hmm yeah that could get tricky..  This is a problem unique to code bases like this one, with the plugin architecture.  Maybe the foremanReact mocks need to start looking more like the actual middleware instead of simple `jest.fn()`s.  Middleware isn't that large anyway.. I don't see a problem with a more fleshed-out mock, or even a mock that's a copy-pasta from the actual middleware.  It's at least worth playing around with.

Reply:
Also Foreman's API and Interval middlewares could eventually become their own NPM packages, in which case you could just import them directly.

Reply:
> Hmm yeah that could get tricky.. This is a problem unique to code bases like this one, with the plugin architecture. Maybe the foremanReact mocks need to start looking more like the actual middleware instead of simple `jest.fn()`s. Middleware isn't that large anyway.. I don't see a problem with a more fleshed-out mock, or even a mock that's a copy-pasta from the actual middleware. It's at least worth playing around with.

I think I'm missing the point then, why mock it and add as middleware if it's currently not being used anywhere?

Reply:
No specific reason, it's just the principle that you want the mock Redux store to resemble the real one as closely as possible.  If the middlewares dispatch some actions that the mock store doesn't, your action list in the mock store won't match with what would happen if it were an actual user interaction.  I suppose it's fine if you're sure that's not happening, but once usage expands it'll be something to think about.

Reply:
gotcha, it does seem like something to think about as we build out the r-t-lib tests more :+1:

Reply:
Now that `foremanReact` can be used for testing, is this resolved now?